src/include/buffer.h

```diff
@@ -317,6 +318,36 @@ class CEPH_BUFFER_API buffer {
 
     };
 
+    class const_iterator : private iterator {
+    public:
+      const_iterator(const iterator& it)
+	: iterator(it) {}
+      const_iterator(const list *l, unsigned o=0)
+	: iterator(const_cast<list*>(l), o) {}
```

Comment:
yes, generally, we don't define a `const_iterator` using a non-const one. "this approach is sort of controversial", i guessed when i posted this pull request.

an alternative way would be use a template to extract the shared parts of these two iterators, and specialize the different bits. but its readability won't be as good as this one.

i will pull together one later on.